ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -446,3 +523,46 @@ private[appendonlydao] final class TransactionsReader(
     mat
   }
 }
+
+private[appendonlydao] object TransactionsReader {
+
+  /** Splits a range of events into equally-sized sub-ranges.
+    *
+    * @param startExclusive The start exclusive of the range to be split
+    * @param endInclusive The end inclusive of the range to be split
+    * @param numberOfChunks The number of desired target sub-ranges
+    * @param minChunkSize Minimum sub-range size.
+    * @return The ordered sequence of sub-ranges with non-overlapping bounds.
+    */
+  private[appendonlydao] def splitRange(
+      startExclusive: Long,
+      endInclusive: Long,
+      numberOfChunks: Int,
+      minChunkSize: Int,
+  ): Seq[EventsRange[Long]] = {
+    val rangeSize = endInclusive - startExclusive
+
+    numberOfChunks match {
+      case 1 =>
+        Seq(EventsRange(startExclusive, endInclusive))
+
+      case invalid if invalid < 1 =>
+        throw new IllegalArgumentException(
+          s"You can only split a range in a strictly positive number of chunks ($numberOfChunks)"
+        )
+
+      case _ if (rangeSize / numberOfChunks.toLong) < minChunkSize.toLong =>
```

Comment:
This condition seems to lead to unintended results e.g. rangeSize = 900, numberOfChunks = 100 will result in a single large range of 900

Reply:
But I guess you're right. Let's make it more defensive and adapt if somehow the parameters are wrongly set.

Reply:
The solution is simple:
effectiveNumChunks = ceil(rangeSize/minChunkSize.toLong)

Reply:
Done